Sources/SwiftDocC/Infrastructure/Workspace/DefaultAvailability.swift

```diff
@@ -36,32 +44,82 @@ public struct DefaultAvailability: Codable, Equatable {
         enum CodingKeys: String, CodingKey {
             case platformName = "name"
             case platformVersion = "version"
+            case unavailable = "unavailable"
+        }
+        
+        /// The diferent availability states that can be declared.
+        /// Unavailable or Available with an introduced version.
+        public enum State: Equatable, Hashable, Encodable {
```

Comment:
Does this _need_ to be public? If this is declared public then changing its cases will be source breaking.

If it can be made `internal` then that would be the preferred solution. 

If that's not possible, I think that we should change the `available` case to `case available(introduced: String, deprecated: String?, obsoleted: String?)` so that we can hopefully avoid some source breaking changes in the future if we add support for more information. We could also introduce a struct and use that as the enum value instead:
```swift
public struct AvailableInfo { 
   public var introducedVersion: String
}

public enum State {
   case unavailable
   case available(AvailableInfo)
}
```

Reply:
Also, I wonder if there a more descriptive type name that we could use here. The word "state" to me implies that it's a value that changes but this value doesn't. If it was an internal type I would maybe name it `Storage` or `Value` or something along those lines instead.

Reply:
I cannot set it to internal because it's used by a public initializer. I like the struct approach tho.

Reply:
We don't need to add a new public initializer with a state argument. There's already an initializer that takes a version string and creates a `State.available` value. All we need is an initializer to create an unavailable value. We could do something like
```
public init(unavailablePlatformName: PlatformName) {
    self.platformName = platformName
    self.state = .unavailable
}
```
and not have to expose the `State` in public API

Reply:
Thx David, resolved